src/dtx/dtx_rpc.c

```diff
@@ -569,6 +570,7 @@ dtx_dti_classify(struct ds_pool *pool, daos_handle_t tree,
 	int	rc = 0;
 	int	i;
 
+	ABT_rwlock_rdlock(pool->sp_lock);
```

Comment:
since you’re accessing pool map within dtx_classify_one(), it's better to add a comment and clarify while taking the lock.